pkg/plugins/policies/meshtimeout/plugin/v1alpha1/plugin.go

```diff
@@ -193,13 +193,23 @@ func applyToGateway(
 		for _, listenerHostname := range listenerInfo.ListenerHostnames {
 			route, ok := gatewayRoutes[listenerHostname.EnvoyRouteName(listenerInfo.Listener.EnvoyListenerName)]
 
-			if conf != nil && ok {
+			if ok {
 				for _, vh := range route.VirtualHosts {
 					for _, r := range vh.Routes {
+						routeConf := getConf(toRules, core_rules.MeshSubset().WithTag(core_rules.RuleMatchesHashTag, r.Name, false))
+						if conf == nil && routeConf == nil {
+							continue
+						}
+						if routeConf == nil && conf != nil {
+							routeConf = conf
+						}
+						if routeConf == nil {
+							continue
+						}
```

Comment:
this seems a bit magic to me, what are we picking here?

Reply:
So the logic is that if there is no config for the specific route we are using config for all routes. 
You might have a configuration for Mesh and MeshHTTPRoute, in case there is no policy for the route, use Mesh one

Reply:
can you add a comment about this? that we fallback